Duplicati/Library/Main/Operation/RestoreHandler.cs

```diff
@@ -105,56 +108,44 @@ public void Run(string[] paths, Library.Utility.IFilter filter = null)
-            m_result.OperationProgressUpdater.UpdatePhase(OperationPhase.Restore_RecreateDatabase);
-
-            using (var tmpdb = new Library.Utility.TempFile())
-            {
-                RecreateDatabaseHandler.NumberedFilterFilelistDelegate filelistfilter = FilterNumberedFilelist(m_options.Time, m_options.Version);
-
-                // Simultaneously with downloading blocklists, we patch as much as we can from the blockvolumes
-                // This prevents repeated downloads, except for cases where the blocklists refer blocks
-                // that have been previously handled. A local blockvolume cache can reduce this issue
-                using (var database = new LocalRestoreDatabase(tmpdb))
+                else
                 {
-                    using (var metadatastorage = new RestoreHandlerMetadataStorage())
-                    {
-                        // TODO: When UpdateMissingBlocksTable is implemented, the localpatcher
-                        // (removed in revision 9ce1e807 ("Remove unused variables and fields") can be activated
-                        // and this will reduce the need for multiple downloads of the same volume
-                        // TODO: This will need some work to preserve the missing block list for use with --fh-dryrun
-                        m_result.RecreateDatabaseResults = new RecreateDatabaseResults(m_result);
-                        using (new Logging.Timer(LOGTAG, "RecreateTempDbForRestore", "Recreate temporary database for restore"))
-                            new RecreateDatabaseHandler(m_backendurl, m_options, (RecreateDatabaseResults)m_result.RecreateDatabaseResults)
-                                .DoRun(database, false, filter, filelistfilter, null);
-
-                        if (!m_options.SkipMetadata)
-                            ApplyStoredMetadata(m_options, metadatastorage);
-                    }
+                    Logging.Log.WriteInformationMessage(LOGTAG, "NoLocalDatabase", "No local database, building a temporary database");
+                    tmpdb = new TempFile();
+                    RecreateDatabaseHandler.NumberedFilterFilelistDelegate filelistfilter = FilterNumberedFilelist(m_options.Time, m_options.Version);
+                    db = new LocalRestoreDatabase(tmpdb);
+                    m_result.RecreateDatabaseResults = new RecreateDatabaseResults(m_result);
+                    using (new Logging.Timer(LOGTAG, "RecreateTempDbForRestore", "Recreate temporary database for restore"))
+                        new RecreateDatabaseHandler(m_backendurl, m_options, (RecreateDatabaseResults)m_result.RecreateDatabaseResults)
+                            .DoRun(db, false, filter, filelistfilter, null);
+
+                    if (!m_options.SkipMetadata)
+                        ApplyStoredMetadata(m_options, new RestoreHandlerMetadataStorage());
+                }
 

```

Comment:
I think this comment is important.
From what I recall, this is a hack from the FE, when restoring without a pre-existing local database.
Maybe the fix is to change the FE to work correctly in this case.

Reply:
Right, this shouldn't have been removed. Reintroduced the comment and the code when the database doesn't exist, which is where the code and comment resided before.